source/includes/library-reference/_config.html.md

```diff
@@ -0,0 +1,109 @@
+# Config Class
+
+Used to manage bonsai configuration environments.
+Config files can be either specified in the users home directory
+or in a local directory. In addition, configuration parameters can be parsed from
+the command line.
+
+Example `~/.bonsai` config file:
+    
+```ini
+[DEFAULT]
+username = admin
+accesskey = None
+profile = dev
+
+[dev]
+url = http://localhost:5000
+username = admin
+accesskey = 00000000-1111-2222-3333-000000000001
+
+[alpha]
+url = https://alpha-api.int.bons.ai
+username = mikest
+```
+
+The `profile` key can be used to switch between different profiles in
+the same configuration file.
+
+## Config(profile)
+
+Constructs a default configuration.
+- `profile` Name of the default profile. Default value is an empty string.
+
+Configurations are stored in `~/.bonsai` and `./.bonsai` configuration files.
+The local configuration file will override settings in the users home directory configuration file.
+    
+```python
+config = Config(sys.argv)
+print(config)
+```
+
+```cpp
+int main(int argc, char** argv) {
+    auto config = Config(argc, argv);
+    std::cout << config << std::endl;
+}
+```
+
+## Config(argc, argv, profile)
+
+Constructs a config by looking in the configuration files and parsing the command line arguments.
+In Python, `argc` is not necessary.
```

Comment:
would be useful to have different variants for Python vs C++ if it is different, no?

Reply:
we should discuss how to do this consistently. there are some tradeoffs to the method we've chosen, am open to revisiting it!